### PR TITLE
DUOS-1104[risk=no]Cancel Election does not update all of the election's votes correctly

### PR DESCRIPTION
**Pull request by JVThomas**
Addresses [DUOS-1104](https://broadworkbench.atlassian.net/browse/DUOS-1104)

PR corrects vote assignments on votes with `type == 'FINAL'` via `updateFinalVote` method within `DatabaseElectionAPI`. Instead of finding a single Final vote, the method will now query for a list of final votes and perform attribute updates based on the election argument. 

**Note:** Method feels like it should belong in a Vote or Election service, and it's only referenced in a single method so moving shouldn't be difficult. I'll write the test up as a follow-up to the feedback.

Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've followed [the instructions](https://github.com/DataBiosphere/consent/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've updated the RC_XXX release ticket with any manual steps required to release this change
- [ ] I've updated the [FISMA documentation](https://github.com/DataBiosphere/consent/blob/develop/CONTRIBUTING.md#fisma-documentation-changes) if I've made any security-related changes, including auth, encryption, or auditing
- [ ] I've updated Swagger to reflect any API changes

In all cases:

- [ ] Get two thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this **unless it's for a hotfix**. In that case, don't delete it!
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Comment by JVThomas** [comment]
Upon further discussion, it seems like the target function, `updateFinalVote` does not belong within an Election update path and should be moved elsewhere. Ideal action is to remove the method and its function call within updateElectionById.

File: src/main/java/org/broadinstitute/consent/http/service/DatabaseElectionAPI.java [hunk 1]
```diff
@@ -611,11 +611,21 @@ private void updateAccessElection(Integer electionId, String type, String status
...
+            for (Vote vote : finalVotes) {
+                vote.setVote(rec.getFinalVote());
+                vote.setUpdateDate(finalVoteDate);
+                vote.setRationale(rec.getFinalRationale());
+                voteDAO.updateVote(
+                    vote.getVote(), vote.getRationale(), vote.getUpdateDate(), vote.getVoteId(),
+                    vote.getIsReminderSent(), vote.getElectionId(), vote.getCreateDate(),
+                    vote.getHasConcerns()
+                );
+            }
```
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/service/DatabaseElectionAPI.java]: In the context of where this is called from, `updateElectionById`, I don't think we should be modifying any votes at all. Vote modification should only ever be done when creating an election (and creating votes for that election), or when updating a vote's value directly through the Vote APIs. My recommendation here is to remove this call altogether. It is named one thing (update final vote) and then queries for a Chairperson vote, making this even more suspect.

**Review by rushtong — CHANGES_REQUESTED** [review]
My suggestion is to remove this method altogether and not update any vote based on an election's status change.

**Review by rushtong — APPROVED** [review]
👍🏽

**Pull request merged**